### ISSUE TITLE
Request: Make releases

### ISSUE THREAD

**Issue by NexAdn** [uowuo/abaddon#34]
So I've been using abaddon for quite some time now and I still like it.

However, I find it hard to follow changes in the software and–since I had to package the software myself–I have trouble following what changes might require an adjustment of the build system (e.g. #29, which removes the need for me to create a wrapper script as a workaround) and–more importantly–when there are notable changes that might require me to update the software, since I don't get notified about updates without following _all_ the commits that happen on master.

Thus, I propose creating releases, e.g. whenever there are notable changes or when some time has gone by without a new release, but new commits (features, bugfixes, changes) have been added to master since the last release. This makes maintenance downstream much easier, since distro maintainers just have to wait for new releases (in GitHub, this is as easy as enabling notifications for new releases) and users don't have to watch for new commits to decide when to re-build their package to get the latest version (which even is a bit counter-intuitive, from a user's perspective).

For development, planning releases with sets of issues to resolve is also a good way to organize in which order issues shall be resolved (e.g. bugfixes for the next patch-level bump, non-breaking changes for the next minor-level bump, breaking changes for the next major-level bump).

**Comment by ouwou** [comment]
yeah it's a good idea and something i was thinking about the other day. i don't really consider this as v1 yet so i figure it might be a good idea starting at say 0.1.0 and incrementing the minor level for something like threads which i just merged or something like #29 and patch being bumped every once in a while with whatever's new if nothing worthy of a minor bump has happened. since this isn't a library or something of the sort idk how the major version would work because "breaking change" isn't that concrete. i think i'd just set it to 1.x.x when i think it deserves it and leave it at that unless i do something drastic. i'll probably make the first release in the next couple days.

**Comment by NexAdn** [comment]
Well, semantic versioning allows any change to happen in minor bumps as long as the major version is 0. And while the major version is 0, the software shall be considered incomplete or in development, so 0.x.y release are currently the way to go.

For major bumps, there might definitely scenarios which cause breaking changes, e.g.

* changing the way the UI works/feels
* changing the config file format completely
* changing CLI paramaters completely

And, after all, you are _not_ bound to semantic versioning. While today almost everyone expects a library to follow semantic versioning, this is _not_ a library. I'd say make up your rules when to do a major bump and the rest can follow semantic versioning as much as is appropriate.

**Comment by NexAdn** [comment]
Another thing from a package mainainter's view: A package maintainer doesn't really care _how_ you make your versions (okay, on some distros like e.g. Gentoo certain versioning schemes make packaging easier for certain libraries, but that's not the case here), as long as you make versions which make it clear when a package maintainer has to bump a new version and thus make users install the package again (now with the new version).

**Comment by ouwou** [comment]
alright well i'll close this since i think i'll start doing releases soon. still not 100% certain how i want to pace out the versions but i think i can figure it out going along.
